Makefile

```diff
@@ -1,40 +1,51 @@
 SHELL = /bin/sh
-PROJECT = graze/silex-trailing-slash-handler
+IMAGE ?= graze/php-alpine
 
-.PHONY: install clean help
-.PHONY: test test-unit test-matrix
+DOCKER_RUN := docker run --rm -it -v $$(pwd):/srv -w /srv ${IMAGE}
+
+.PHONY: install update update-lowest help
+.PHONY: test test-unit test-matrix test-matrix-lowest
 
 .SILENT: help
 
-install: ## Download the depenedencies then build the image :rocket:.
-	make 'composer-install --optimize-autoloader --ignore-platform-reqs'
-	docker build --tag ${PROJECT}:latest .
+install: ## Download the dependencies then build the image :rocket:.
+	${MAKE} 'composer-install -o --prefer-dist'
+
+update: ## Update the dependencies
+	${MAKE} 'composer-update -o --prefer-dist'
+
+update-lowest: ## Update to the lowest stable dependencies
+	${MAKE} 'composer-update -o --prefer-dist --prefer-lowest --prefer-stable'
 
 composer-%: ## Run a composer command, `make "composer-<command> [...]"`.
 	docker run -it --rm \
-	-v $$(pwd):/app \
-	-v ~/.composer:/root/composer \
-	-v ~/.ssh:/root/.ssh:ro \
-	composer/composer --ansi --no-interaction $*
+        -v $$(pwd):/app \
+        -v ~/.composer:/tmp \
+        -v ~/.ssh:/root/.ssh:ro \
+        composer --ansi --no-interaction $*
```

Comment:
Consider changing `$*` to `"$@"` to take into account spaces and handling quotes sentences as single values.

Reply:
`$*` has a sort of special meaning here (as in what was the `%` bit in the command):

so `make 'composer-install -o'` will produce `composer --ansi --no-interaction install -o` whereas adding `"$*"` or `"$@"` puts quotes around it so it calls `composer --ansi --no-interaction "install -o"`

Reply:
Fair enough. I forgot the context. This isn't the shell!